src/lib/y2network/config_reader/autoyast_routing.rb

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/routing_table"
+require "y2network/routing"
+require "y2network/route"
+
+module Y2Network
+  module ConfigReader
+    # This class is responsible of importing the AutoYast routing section
+    class AutoyastRouting
+      # @return [AutoinstProfile::RoutingSection]
+      attr_reader :section
+
+      # @param section [AutoinstProfile::RoutingSection]
+      def initialize(section)
+        @section = section
+      end
+
+      # Creates a new {Routing} config from the imported profile routing section
+      #
+      # @return [Routing] the imported {Routing} config
+      def config
+        tables = section.routes ? [Y2Network::RoutingTable.new(build_routes)] : []
+        Y2Network::Routing.new(tables:       tables,
+                               forward_ipv4: !!section.ipv4_forward,
+                               forward_ipv6: !!section.ipv6_forward)
+      end
+
+    private
+
+      # Build a route given a route section
+      #
+      # @return [Array<Route>]
+      def build_routes
+        section.routes.map do |route_section|
+          Y2Network::Route.new(to:        route_section.destination || :default,
```

Comment:
how looks default route? It use nil there or "-" or "default" ?

Reply:
In the profile it should be "default" or IP[/prefix]:

```xml
 <route>
        <destination>default</destination>
        <device>eth0</device>
        <gateway>192.168.1.1</gateway>
        <netmask>-</netmask>
 </route>
```

So, the init_from_hashes for RouteSection is still pending. Will add it

Reply:
I prepared it for being able to do

```xml
  <route>
    <destination>default</destination>
    <gateway>192.168.1.1</gateway>
  </route>
```
https://github.com/yast/yast-network/pull/755/files#diff-89a18cae535af72c8ff5a12849265230R268

But destination has to be always declared

Reply:
ok, so in this line "to" will be `"default"`, which is not intended, it should be `:default`

Reply:
Yep, in the `RouteSection` object it is "default" and should be transformed to a symbol or to an `IPAddr` in case it is not "default"

Reply:
I have seen in different places that an assumption of "-" for default was used and that is not correct.

https://github.com/yast/yast-network/pull/755/files#diff-2072ed0d82b988333aa0f3e3e802570cR55 (fixed)

See original validation:

https://github.com/yast/yast-network/blob/master/src/include/network/services/routing.rb#L189

So, take it into account in widgets

https://github.com/yast/yast-network/blob/network-ng/src/lib/y2network/widgets/destination.rb#L34
https://github.com/yast/yast-network/blob/network-ng/src/lib/y2network/widgets/destination.rb#L54

I will propose the fix as part of the PBI if you want although probably it will use route.default? instead of handling the content for that time.

Reply:
widgets are already modified in my branch